core/lib.rs

```diff
@@ -428,7 +424,56 @@ impl Database {
         self.open_flags.contains(OpenFlags::ReadOnly)
     }
 
-    fn init_pager(&self, page_size: Option<usize>) -> Result<Pager> {
+    /// If we do not have a physical WAL file, but we know the database file is initialized on disk,
+    /// we need to read the page_size from the database header.
+    fn read_page_size_from_db_header(&self) -> Result<PageSize> {
+        turso_assert!(
+            self.db_state.is_initialized(),
+            "read_page_size_from_db_header called on uninitialized database"
+        );
+        let buf = Arc::new(Buffer::new_temporary(PageSize::MIN as usize)); // this needs to be a block multiple for O_DIRECT
+        let c = Completion::new_read(buf.clone(), move |_buf, _| {});
+        let c = self.db_file.read_header(c)?;
+        self.io.wait_for_completion(c)?;
+        let page_size = u16::from_be_bytes(buf.as_slice()[16..18].try_into().unwrap());
+        Ok(PageSize::new(page_size as u32).unwrap_or_else(|| {
```

Comment:
Hm...does `PageSize::new` constructor do what we need here? It seems like it convert page_size logical value to the physical U16 representation - but we actually want reverse action here (I guess the only difference is for 65536 page size here - which represented as `0x00 0x01` in binary format)

Reply:
Yes, but constructor expect that you will call it with the actual size, right? But in case of 65536 page size - we will read `0x00 0x01` from the disk and call constructor with the `page_size = 1`. No?

Reply:
I added an explicit exception to the constructor of `PageSize` for the value `1`

Reply:
Hm...is it OK? I mean that the semantic of constructor is not clear to me. Is it accepting binary representation of the page_size or the "logical" value of the page size? For example, if `PageSize` struct used as the "validation" barrier in the `PRAGMA page_size=N` - then turso-db can accidentally accept `PRAGMA page_size=1` commands which is not what we want it to do.

Reply:
Actually, the WAL has a 4-byte page size, so this is only relevant for the DB header